src/ui/screens/feed-settings/FeedSettingsContainer.ts

```diff
@@ -1,13 +1,41 @@
 import { connect } from 'react-redux';
-import { StateProps, FeedSettingsScreen } from './FeedSettingsScreen';
-import { AppState } from '../../../reducers';
+import { StateProps, DispatchProps, FeedSettingsScreen } from './FeedSettingsScreen';
+import { AppState } from '../../../reducers/AppState';
+import { LocalFeed } from '../../../social/api';
+import { Actions } from '../../../actions/Actions';
+import { emptyPostCommandLog } from '../../../social/api';
+
+const emptyFeed: LocalFeed = {
+    name: '',
+    url: '',
+    feedUrl: '',
+    favicon: '',
+    isSyncing: false,
+    authorImage: {},
+    posts: [],
+    postCommandLog: emptyPostCommandLog,
+    autoShare: false,
+};
 
 const mapStateToProps = (state: AppState, ownProps: { navigation: any }): StateProps => {
+    const paramFeed = ownProps.navigation.state.params.feed;
```

Comment:
Here the story is that if the feed comes from the navigation parameter then it won't be updated when something changes in the component, so basically it's read-only.

Therefore we lookup the actual feed from the state by the feedUrl which is always supposed to work. In order to avoid having to pass `LocalFeed | undefined` to the component and having many null-checks I found this to be a good enough workaround.

Reply:
Can't it be solved by `!`?